pkg/lifecycle/service.go

```diff
@@ -119,31 +184,31 @@ func (s *Service) Start(
 // instead the context for all nodes will be canceled which causes them to stop
 // running as soon as possible.
 func (s *Service) Stop(ctx context.Context, pipelineID string, force bool) error {
-	pl, err := s.Get(ctx, pipelineID)
-	if err != nil {
-		return err
+	rp, ok := s.runningPipelines[pipelineID]
```

Comment:
Watch out, we need to lock the mutex before accessing `s.runningPipelines`.

Reply:
Maybe it would be easier and safer to use a `sync.Map`?

Reply:
I'm not a fan, since we then need to do type assertions 😕 That said, we could provide our own safe implementation in `csync` that uses generics, but not sure it's needed in this PR.

Reply:
Cool, didn't know that about types (I didn't use it myself yet). :+1: So yeah, locks be it.:)

Reply:
Man, nowadays with AI's help such a struct is implemented and tested insanely fast. https://github.com/ConduitIO/conduit-commons/pull/116 🙃

Reply:
https://github.com/ConduitIO/conduit/pull/1835/commits/3596fb82428f71d8d990aed272d2407fd2344157